src/main/java/com/puppycrawl/tools/checkstyle/checks/design/OneTopLevelClassCheck.java

```diff
@@ -98,8 +99,9 @@
      */
     private boolean publicTypeFound;
 
-    /** Mapping between type names and line numbers of the type declarations.*/
-    private final SortedMap<Integer, String> lineNumberTypeMap = new TreeMap<>();
+    /** Mapping between type names and DetailAST nodes of the type declarations.*/
+    private final SortedMap<DetailAST, String> lineNumberTypeMap =
+            new TreeMap<>(Comparator.comparingInt(DetailAST::getLineNo));
```

Comment:
I doubt it should be like this. In very rare case, e.g.
`enum A {}; enum B{}` no violation will be reported. 
Please change it.

Reply:
I changed the comparator to also check the column number if the line number is equal. That should be good enough to ensure the nodes are unique and still in order.